drivers/sensor/bmp085/bmp085.c

```diff
@@ -0,0 +1,328 @@
+	LOG_INF("bmpAC3: %d/0x%X", bmp085->bmpAC3, bmp085->bmpAC3);
+
+	bmp085->bmpAC4 = buf[7] | (buf[6] << 8);
+	LOG_INF("bmpAC4: %d/0x%X", bmp085->bmpAC4, bmp085->bmpAC4);
+
+	bmp085->bmpAC5 = buf[9] | (buf[8] << 8);
+	LOG_INF("bmpAC5: %d/0x%X", bmp085->bmpAC5, bmp085->bmpAC5);
+
+	bmp085->bmpAC6 = buf[11] | (buf[10] << 8);
+	LOG_INF("bmpAC6: %d/0x%X", bmp085->bmpAC6, bmp085->bmpAC6);
+
+	bmp085->bmpB1 = buf[13] | (buf[12] << 8);
+	LOG_INF("bmpB1: %d/0x%X", bmp085->bmpB1, bmp085->bmpB1);
+
+	bmp085->bmpB2 = buf[15] | (buf[14] << 8);
+	LOG_INF("bmpB2: %d/0x%X", bmp085->bmpB2, bmp085->bmpB2);
+
+	bmp085->bmpMB = buf[17] | (buf[16] << 8);
+	LOG_INF("bmpMB: %d/0x%X", bmp085->bmpMB, bmp085->bmpMB);
+
+	bmp085->bmpMC = buf[19] | (buf[18] << 8);
+	LOG_INF("bmpMC: %d/0x%X", bmp085->bmpMC, bmp085->bmpMC);
+
+	bmp085->bmpMD = buf[21] | (buf[20] << 8);
+	LOG_INF("bmpMD: %d/0x%X", bmp085->bmpMD, bmp085->bmpMD);
+
+	return 0;
+}
+
+static inline int bmp085_bus_config(const struct device *dev)
+{
+	const struct bmp085_device_config *dev_cfg = dev->config;
+	uint32_t i2c_cfg;
+
+	i2c_cfg = I2C_MODE_CONTROLLER | I2C_SPEED_SET(BMP085_I2C_BUS_SPEED);
+
+	return i2c_configure(dev_cfg->i2c.bus, i2c_cfg);
+}
+
+int bmp085_read(const struct device *dev, uint8_t reg_addr, uint8_t *data, uint8_t len)
+{
+	const struct bmp085_device_config *dev_cfg = dev->config;
+	int ret = 0;
+
+	bmp085_bus_config(dev);
+
+	if (i2c_burst_read_dt(&dev_cfg->i2c, reg_addr, data, len) < 0) {
+		ret = -EIO;
+	}
+
+	return ret;
+}
+
+int bmp085_read_byte(const struct device *dev, uint8_t reg_addr, uint8_t *byte)
+{
+	return bmp085_read(dev, reg_addr, byte, 1);
+}
+
+static int bmp085_write(const struct device *dev, uint8_t reg_addr, uint8_t *data, uint8_t len)
+{
+	const struct bmp085_device_config *dev_cfg = dev->config;
+	int ret = 0;
+
+	bmp085_bus_config(dev);
+
+	if (i2c_burst_write_dt(&dev_cfg->i2c, reg_addr, data, len) < 0) {
+		ret = -EIO;
+	}
+
+	return ret;
+}
+
+int bmp085_write_byte(const struct device *dev, uint8_t reg_addr, uint8_t byte)
+{
+	return bmp085_write(dev, reg_addr, &byte, 1);
+}
+
+static int bmp085_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	int ret = 0;
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL);
+
+	ret |= bmp085_read_temp(dev);
+	ret |= bmp085_read_press(dev, BMP085_OVERSAMPLING);
+
+	return ret;
+}
+
+static int bmp085_channel_get(const struct device *dev, enum sensor_channel chan,
+			      struct sensor_value *val)
+{
+	struct bmp085_device_data *bmp085 = dev->data;
+
+	switch (chan) {
+	case SENSOR_CHAN_AMBIENT_TEMP:
+		val->val1 = bmp085->temp / 10;
+		val->val2 = bmp085->temp % 10;
+		break;
+	case SENSOR_CHAN_PRESS:
+		val->val1 = bmp085->press / 1000;
+		val->val2 = bmp085->press % 1000;
+		break;
+	default:
+		return -ENOTSUP;
+	}
+	return 0;
+}
+
+static const struct sensor_driver_api bmp085_api = {
+	.sample_fetch = bmp085_sample_fetch,
+	.channel_get = bmp085_channel_get,
+};
+
+int bmp085_init(const struct device *dev)
```

Comment:
```suggestion
static int bmp085_init(const struct device *dev)
```
Please check that all functions in this file are static